library/Vanilla/PageScraper.php

```diff
@@ -190,4 +187,107 @@ private function userAgent() {
         $version = defined('APPLICATION_VERSION') ? APPLICATION_VERSION : '0.0';
         return "Vanilla/{$version}";
     }
+
+    /**
+     * Create the DOM from an HTML string.
+     *
+     * @param string $html The HTML to create the body from.
+     * @return DOMDocument Returns the document.
+     * @throws Exception Throws an exception if the HTML is so malformed that it couldn't be parsed.
+     */
+    private function createDom(string $html): DOMDocument {
+        // Add charset information for HTML 5 pages.
+        // https://stackoverflow.com/questions/39148170/utf-8-with-php-domdocument-loadhtml#39148511
+        if (!preg_match('`^\s*<?xml`', $html) && $encoding = mb_detect_encoding($html)) {
```

Comment:
do we really need t his here? we have another detection inside fixDocumentEncoding function